patches/minecraft/net/minecraft/world/level/block/entity/BlockEntityType.java.patch

```diff
@@ -0,0 +1,22 @@
+--- a/net/minecraft/world/level/block/entity/BlockEntityType.java
++++ b/net/minecraft/world/level/block/entity/BlockEntityType.java
+@@ -85,7 +_,7 @@
+    }
+ 
+    public boolean m_155262_(BlockState p_155263_) {
+-      return this.f_58915_.contains(p_155263_.m_60734_());
++      return this.f_58915_.contains(p_155263_.m_60734_()) || additionalValidBlocks.contains(net.minecraftforge.registries.ForgeRegistries.BLOCKS.getResourceKey(p_155263_.m_60734_()));
+    }
+ 
+    @Nullable
+@@ -115,5 +_,10 @@
+       public BlockEntityType<T> m_58966_(Type<?> p_58967_) {
+          return new BlockEntityType<>(this.f_58959_, this.f_58960_, p_58967_);
+       }
++   }
++   // FORGE START. INTERNAL USE ONLY!
++   private final java.util.Set<net.minecraft.resources.ResourceKey<Block>> additionalValidBlocks = new java.util.HashSet<>();
++   public static void fireAddBlockEvent() {
++       net.minecraftforge.common.ForgeHooks.onBlockEntityBlocks().forEach((type, blocks) -> type.additionalValidBlocks.addAll(blocks));
```

Comment:
Just because this is a patch does not mean you cannot use whitespace so long as it doesn't increase the required number of context lines. In this case, I would recommend a newline between the ending bracket and forge start comment along with a newline between the field and method.